packages/jest-runtime/src/index.ts

```diff
@@ -636,7 +636,11 @@ export default class Runtime {
 
     const module = await this.loadEsmModule(modulePath, query);
 
-    return this.linkAndEvaluateModule(module);
+    const evaluatedModule = await this.linkAndEvaluateModule(module);
+
+    return evaluatedModule instanceof SourceTextModule
+      ? evaluatedModule.namespace.default
```

Comment:
I have no clue what to do here, but I found this method sometimes returns `vm.SourceTextModule` as result.

Reply:
There is no "result" of executing a module. Do we need access to evaluated module? We haven't needed that before... We might need an `unstable_importInternalModule` which essentially does what you do here (wait for [`evaluatedModule.status === 'evaluated'`](https://nodejs.org/api/vm.html#modulestatus), then fetch its default export).

Essentially faking an `import()` call

Reply:
If i didn't get it wrong, the result is used by this `localRequire` call to support ESM resolvers:

```js
const custom: SnapshotResolver = await localRequire(
    snapshotResolverPath,
    true,
 );
```